ui/app/src/nh-config/create-dimension-form.ts

```diff
@@ -0,0 +1,382 @@
+import { AppInfo, DnaHash, EntryHash } from "@holochain/client";
+import { contextProvided } from "@lit-labs/context";
+import { NHButton, NHCard, NHComponentShoelace } from "@neighbourhoods/design-system-components";
+import { html, css, CSSResult, PropertyValueMap } from "lit";
+import { matrixContext, weGroupContext } from "../context";
+import { MatrixStore } from "../matrix-store";
+import { SlInput, SlRange } from "@scoped-elements/shoelace";
+import { object, string, boolean, number } from 'yup';
+import { Dimension, Range, RangeKind, SensemakerStore } from "@neighbourhoods/client";
+import { property, query, state } from "lit/decorators.js";
+
+const MIN_RANGE = -1_000_000;
+const MAX_RANGE = 1_000_000;
+
+export default class CreateDimension extends NHComponentShoelace {
+  @property()
+  sensemakerStore!: SensemakerStore;
+  
+  @property()
+  dimensionType!: "input" | "output";
+
+  @property()
+  methodComputation!: "AVG" | "SUM";
+  
+  @contextProvided({ context: matrixContext, subscribe: true })
+  _matrixStore!: MatrixStore;
+
+  @contextProvided({ context: weGroupContext, subscribe: true })
+  weGroupId!: DnaHash;
+
+  _dimensionSchema = object({
+    name: string().min(1, "Must be at least 1 characters").required(),
+    computed: boolean().required(),
+  });
+  
+  _currentMinRange : number = 0;
+  _dimensionRangeSchema = () => object({
+    min: number().min(MIN_RANGE, "Must be at least " + MIN_RANGE).required(),
+    max: number().min(this._currentMinRange + 1, "Must be > " + this._currentMinRange).max(MAX_RANGE, "Must be at most " + MAX_RANGE),
```

Comment:
"The higher extent of this range cannot be higher than " + the maximum range for the given data type (see my long comment on numeric types).